src/vs/workbench/services/views/browser/viewDescriptorService.ts

```diff
@@ -230,7 +236,7 @@ export class ViewDescriptorService extends Disposable implements IViewDescriptor
 		this._register(this.viewsRegistry.onViewsRegistered(({ views, viewContainer }) => this.onDidRegisterViews(views, viewContainer)));
 		this._register(this.viewsRegistry.onViewsDeregistered(({ views, viewContainer }) => this.onDidDeregisterViews(views, viewContainer)));
 
-		this._register(this.viewsRegistry.onDidChangeContainer(({ views, from, to }) => { this.removeViews(from, views); this.addViews(to, views); }));
+		this._register(this.viewsRegistry.onDidChangeContainer(({ views, from, to }) => { this.removeViews(from, views); this.addViews(to, views); this._onDidChangeContainer.fire({ views, from, to }); }));
```

Comment:
I think we shall also check location change here even though currently location change is not yet supported in  registry. This should be ready for future if it is allowed.

I would recommend to have a single method that can be reused when views are moved in registry or in cache.

```ts
this.removeViews(from, views);
 this.addViews(to, views);

  const oldLocation = this.viewContainersRegistry.getViewContainerLocation(from)!;
 const newLocation = this.viewContainersRegistry.getViewContainerLocation(to)!;

  if (oldLocation !== newLocation) {
 	this._onDidChangeLocation.fire({ views, from: oldLocation, to: newLocation });
 }

 this._onDidChangeContainer.fire({ views, from, to });
```

Reply:
@sandy081 thanks, will take care of this.